packages/sdk-codegen/src/sdkModels.ts

```diff
@@ -304,6 +304,16 @@ export interface IType extends ISymbol {
    */
   properties: PropertyList
 
+  /**
+   * key/value collection of required properties for this type
+   */
+  requiredProperties: PropertyList
```

Comment:
Do you need this as a list, or if it's just for natural order iteration, it could be `IProperty[]`

Reply:
I wanted it to be the same type as `properties` - `ICodeGen.typeProperties` returns a `IProperty[]` because that's all `declareType` cares about is natural order iteration